lib/BackbeatConsumer.js

```diff
@@ -587,22 +592,24 @@ class BackbeatConsumer extends EventEmitter {
                     const logger = this._consumer.isConnected() ? this._log.error : this._log.info;
                     logger.bind(this._log)('rdkafka.unassign failed', { e: e.toString(), assignment });
                 }
+
+                this.emit('unassign', status);
             });
 
             if (!this._processingQueue || this._processingQueue.idle()) {
-                unassign('idle');
+                unassign(UNASSIGN_STATUS.IDLE);
                 return;
             }
 
-            this._processingQueue.drain = () => unassign('drained');
+            this._processingQueue.drain = () => unassign(UNASSIGN_STATUS.DRAINED);
```

Comment:
shouldn't we stick to the 'empty' callback?
the queue should never drain (or fill) unless we go through this revoke-ation process anyway, so it seems safer to not emit the signal...